lib/internal/modules/esm/formats.js

```diff
@@ -35,7 +38,32 @@ function mimeToFormat(mime) {
   return null;
 }
 
+/**
+ * For extensionless files in a `module` package scope, or a default `module` scope enabled by the `--experimental-type`
+ * flag, we check the file contents to disambiguate between ES module JavaScript and Wasm.
+ * We do this by taking advantage of the fact that all Wasm files start with the header `0x00 0x61 0x73 0x6d` (`_asm`).
+ * @param {URL} url
+ */
+function getFormatOfExtensionlessFile(url) {
+  if (!experimentalWasmModules) { return 'module'; }
+
+  const magic = new Uint8Array(4);
+  let fd;
+  try {
+    fd = openSync(url);
+    readSync(fd, magic, 0, 4); // Only read the first four bytes
+    if (magic[0] === 0x00 && magic[1] === 0x61 && magic[2] === 0x73 && magic[3] === 0x6d) {
+      return 'wasm';
+    }
+  } finally {
+    if (fd) { closeSync(fd); }
```

Comment:
```suggestion
    if (fd !== undefined) closeSync(fd);
```
Because `fd` can be zero.